### PR TITLE
RELATED: RAIL-3794 - Plugin API tweaks / cleanup

### PR DESCRIPTION
**Pull request by lupko**
...

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5892/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7666/ : SUCCESS in 15m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5893/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7667/ : SUCCESS in 20m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5895/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7671/ : SUCCESS in 16m 07s

File: libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx [hunk 1]
```diff
@@ -272,27 +272,34 @@ export class DashboardLoader implements IDashboardLoader {
...
+        const { plugin, parameters } = loadedPlugin;
+
         try {
             if (plugin.onPluginLoaded) {
                 // eslint-disable-next-line no-console
                 console.debug(`Calling onPluginLoaded on ${plugin.displayName}...`);
-                plugin.onPluginLoaded(ctx, parameters);
+                const loadPromise = plugin.onPluginLoaded(ctx, parameters);
+
+                // just in case someone forgets to return Promise.resolve()
```
Comment (no23reason) [inline libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx:299]: I think we can just do
```ts
await plugin.onPluginLoaded(ctx, parameters);
```
you can await anything, even `undefined` so no need for the if. Or am I missing something?

Comment (lupko) [reply libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5903/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7682/ : SUCCESS in 16m 59s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1702/ : SUCCESS in 1m 04s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1983/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/780/ : SUCCESS in 16m 38s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7683/ : SUCCESS in 17m 03s

**Pull request merged**